tests/commons.rs

```diff
@@ -83,6 +83,13 @@ impl TestCase for TestParamsInOut {
 	}
 
 	async fn processor<W: AsyncWrite + Unpin + Send>(request: Arc<Request<W>>) -> RequestResult {
+		let mut keys: Vec<String> = request.get_param_keys().unwrap().map(|k| k.clone()).collect();
+		keys.sort();
+
+		assert_eq!(keys.len(), 2);
+		assert_eq!(keys[0], "server_port");
```

Comment:
This test will randomly fail because the order of keys is not guaranteed by rusts [`keys()`](https://doc.rust-lang.org/std/collections/hash_map/struct.HashMap.html#method.keys) method.

Reply:
Good point - I was worried on this same issue.  For this reason I make a copy of the keys in a mutable vector, and then call `keys.sort()` to sort the keys in alphabetical order.

Reply:
Thank you for pointing this out. I've overlooked the `sort`. This should work flawless.